### PR TITLE
Story/ccls 1715 address search results screen

### PR DESCRIPTION
**Pull request by PhilDigitalJustice**
includes:

- CCLS-1715 Address search results screen
- CCLS-1722 Address search integration with ordinance survey
- Updated readme
- Added os api wiremock

### CODE REVIEW

File: src/main/java/uk/gov/laa/ccms/caab/service/AddressService.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  private final ClientAddressResultDisplayMapper clientAddressResultDisplayMapper;
+
+
+  /**
+   * Fetches addresses based on the given postcode by using the Ordinance Survey API.
+   * Maps the API response to ClientAddressResultsDisplay format.
+   *
+   * @param postcode The postcode to search for.
+   * @return ClientAddressResultsDisplay containing a list of addresses associated with the given
+   *         postcode.Returns an empty ClientAddressResultsDisplay if no results are found.
```
Comment (keith-h) [inline src/main/java/uk/gov/laa/ccms/caab/service/AddressService.java:34]: some weird formatting here

File: src/test/java/uk/gov/laa/ccms/caab/bean/validators/client/ClientAddressDetailsFindAddressValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+      clientDetails.setAddressLine1(value);
+    } else if (field.equals("addressLine2")) {
+      clientDetails.setAddressLine2(value);
+    } else if (field.equals("cityTown")) {
+      clientDetails.setCityTown(value);
+    } else if (field.equals("county")) {
+      clientDetails.setCounty(value);
+    }
+
+    clientAddressDetailsFindAddressValidator.validate(clientDetails, errors);
```
Comment (keith-h) [inline src/test/java/uk/gov/laa/ccms/caab/bean/validators/client/ClientAddressDetailsFindAddressValidatorTest.java:114]: should we also be asserting how many errors we're expecting, or that too much detail for these tests?

File: src/test/java/uk/gov/laa/ccms/caab/bean/validators/client/ClientAddressSearchValidatorTest.java [hunk 3]
```diff
@@ -46,83 +40,16 @@ public void supports_ReturnsFalseForOtherClasses() {
...
-    assertFalse(errors.hasErrors());
-  }
-
-  @ParameterizedTest
-  @NullAndEmptySource
-  public void validate_countryRequired(String country) {
-    clientDetails.setCountry(country);
+  public void validate_errors() {
     clientAddressSearchValidator.validate(clientDetails, errors);
     assertTrue(errors.hasErrors());
```
Comment (keith-h) [inline src/test/java/uk/gov/laa/ccms/caab/bean/validators/client/ClientAddressSearchValidatorTest.java:45]: number of errors?

File: src/test/java/uk/gov/laa/ccms/caab/bean/validators/client/ClientAddressSearchValidatorTest.java [hunk 4]
```diff
@@ -46,83 +40,16 @@ public void supports_ReturnsFalseForOtherClasses() {
...
-      clientDetails.setCityTown(value);
-    } else if (field.equals("county")) {
-      clientDetails.setCounty(value);
-    }
-
+  public void validate() {
+    clientDetails.setUprn("12345");
     clientAddressSearchValidator.validate(clientDetails, errors);
-    assertTrue(errors.hasErrors());
+    assertFalse(errors.hasErrors());
```
Comment (keith-h) [inline src/test/java/uk/gov/laa/ccms/caab/bean/validators/client/ClientAddressSearchValidatorTest.java:52]: and here

File: src/main/java/uk/gov/laa/ccms/caab/service/AddressService.java (hunk 1 above)
Comment (PhilDigitalJustice) [reply src/main/java/uk/gov/laa/ccms/caab/service/AddressService.java:34]: A return need to be tabbed like that, to pass checkstyle.

File: src/test/java/uk/gov/laa/ccms/caab/bean/validators/client/ClientAddressDetailsFindAddressValidatorTest.java (hunk 2 above)
Comment (PhilDigitalJustice) [reply src/test/java/uk/gov/laa/ccms/caab/bean/validators/client/ClientAddressDetailsFindAddressValidatorTest.java:114]: makes sense to include that

File: src/test/java/uk/gov/laa/ccms/caab/bean/validators/client/ClientAddressSearchValidatorTest.java (hunk 3 above)
Comment (PhilDigitalJustice) [reply src/test/java/uk/gov/laa/ccms/caab/bean/validators/client/ClientAddressSearchValidatorTest.java:45]: will add count here

Comment (PhilDigitalJustice) [reply src/test/java/uk/gov/laa/ccms/caab/bean/validators/client/ClientAddressSearchValidatorTest.java:46]: amended

File: src/test/java/uk/gov/laa/ccms/caab/bean/validators/client/ClientAddressDetailsFindAddressValidatorTest.java (hunk 2 above)
Comment (PhilDigitalJustice) [reply src/test/java/uk/gov/laa/ccms/caab/bean/validators/client/ClientAddressDetailsFindAddressValidatorTest.java:119]: amended

**Pull request merged**